pkg/podgrouper/podgrouper/plugins/grove/grove_grouper.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+package grove
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/NVIDIA/KAI-scheduler/pkg/podgrouper/podgroup"
+	"github.com/NVIDIA/KAI-scheduler/pkg/podgrouper/podgrouper/plugins/defaultgrouper"
+)
+
+const (
+	podGangNameKey = "grove.io/podgang"
+)
+
+type GroveGrouper struct {
+	client client.Client
+	*defaultgrouper.DefaultGrouper
+}
+
+func NewGroveGrouper(client client.Client, defaultGrouper *defaultgrouper.DefaultGrouper) *GroveGrouper {
+	return &GroveGrouper{
+		client:         client,
+		DefaultGrouper: defaultGrouper,
+	}
+}
+
+func (gg *GroveGrouper) Name() string {
+	return "Grove Grouper"
+}
+
+// +kubebuilder:rbac:groups=grove.io,resources=podgangset,verbs=get;list;watch
+// +kubebuilder:rbac:groups=grove.io,resources=podgangset/finalizers,verbs=patch;update;create
+// +kubebuilder:rbac:groups=scheduler.grove.io,resources=podgangs,verbs=get;list;watch
+// +kubebuilder:rbac:groups=scheduler.grove.io,resources=podgangs/finalizers,verbs=patch;update;create
+
+func (gg *GroveGrouper) GetPodGroupMetadata(
+	_ *unstructured.Unstructured, pod *v1.Pod, _ ...*metav1.PartialObjectMetadata,
+) (*podgroup.Metadata, error) {
+	if len(pod.Spec.SchedulingGates) > 0 {
+		return nil, fmt.Errorf("Ignoring pod %s/%s due to scheduling gates: %v",
+			pod.Namespace, pod.Name, pod.Spec.SchedulingGates)
+	}
```

Comment:
This part is already covered in the scheduler.
https://github.com/NVIDIA/KAI-Scheduler/blob/b66691d1efd35b0ea04fd0fe5232742714b784e4/pkg/scheduler/api/pod_info/pod_info.go#L332

I think you can safely create a podgroup and it won't be scheduled as long as it has a gated pod.